server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -111,19 +116,29 @@ private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<
             .unreadMessages(mailboxCounters.getUnseen())
             .totalMessages(mailboxCounters.getCount())
             .sortOrder(SortOrder.getSortOrder(role))
-            .sharedWith(Rights.fromACL(metaData.getACL()))
-            .namespace(getNamespace(mailboxPath, mailboxSession))
+            .sharedWith(rights)
+            .mayAddItems(rights.mayAddItems(username).orElse(isOwner))
+            .mayCreateChild(rights.mayCreateChild(username).orElse(isOwner))
+            .mayDelete(rights.mayDelete(username).orElse(isOwner))
+            .mayReadItems(rights.mayReadItems(username).orElse(isOwner))
+            .mayRemoveItems(rights.mayRemoveItems(username).orElse(isOwner))
+            .mayRename(rights.mayRename(username).orElse(isOwner))
```

Comment:
rights should have already resolved the owner in the manager layer, don't you think ? how is this isOwner useful ?

Reply:
Well, because the owner "don't have Rights entries" :trollface:

Reply:
yes but what about adding him the rights at the manager level?

Reply:
It's not enougth.

We need some kind of "special" JMAP handling, because we can not express "mayDelete mayRename" with JMAP logic. Hence, we can not resolve owner rights on top of JMAP rights.

Reply:
ok, it took me a while to understand, but I got it.
To explain it differently: if you add owner rights into MessageManager, you cannot express mayDelete and mayRename, because they should not be on this mailbox but on the parent one.
So when you get the rights at the JMAP level, you don't know if mayDelete and mayRename are valid, and so you cannot put them to true when the user is the owner.